.github/workflows/PR.yml

```
@@ -84,7 +90,25 @@ jobs:
         OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
         CO_API_KEY: ${{ secrets.CO_API_KEY }}
         CE_LOG_FILENAME: e2e.log
-      run: poetry run pytest -n 3 --dist loadscope --html=report_e2e.html --self-contained-html tests/e2e
+      run: |
+        run_id=$(uuidgen | tr -d '-' | tr '[:upper:]' '[:lower:]')
+        poetry run pytest -n 3 --dist loadscope --testrunuid $run_id --html=report_e2e.html --self-contained-html tests/e2e
+        echo "Test Run ID: ${run_id}"
+        echo "run_id=${run_id}" >> $GITHUB_OUTPUT
+    - name: Cleanup indexes
+      if: failure() && github.event_name == 'merge_group'
```

```suggestion
      if: always() && github.event_name == 'merge_group'
```
Let's not take any chances. The added time for successful tests is probably negligible.